### ISSUE TITLE
Feature Request: Automatically Open readme.md Files

### ISSUE THREAD

**Issue by CKraniak** [NuGet/Home#10353]
I'm not exactly sure where to put this, and the prompt text doesn't seem to be appropriate for this.

In Nuget.Client:

I would like `readme.md` files to automatically open the same way that `readme.txt` files do.

Looking at the code, this appears it would need:
* a new constant in `src/NuGet.Core/NuGet.PackageManagement/Utility/Constants.cs`
* modifying `OpenReadmeFile()` in `src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs`
  * I think I would factor out computing the directory
  * As it is now, the relevant changes would need to happen on lines 3326 and 3335, I think
  * In both places, immediately check `File.Exists(readmeFilePath)` and if `false` set `readmeFilePath` to point at the `.md` instead
    * Not that great of a solution, I think I'd prefer the refactor
* Define the behavior for the case when they both exist to prefer opening `readme.txt`

I might rather have `ReadmeFileName` be a list of names (and renamed to `ReadmeFileNames`) and cycle through the list, but that requires more work for dubious gains.

I am not interested in signing a contributor license agreement at this time, otherwise I would consider just doing this myself.

**Comment by chrisraygill** [comment]
Hi! We currently have a [spec](https://github.com/NuGet/Home/wiki/Embedding-and-displaying-NuGet-READMEs) to allow for embedding Readme.md files in NuGet packages, rendering them on NuGet.org, and including a link in the VS Package Manager package details pane.

Do you feel it is important that we also support automatically opening the a Readme.md file on package install if it's accessible through a link in the package details pane?

Feel free to also leave feedback on the spec itself for any other important scenarios you feel we might be missing.

**Comment by CKraniak** [comment]
> Do you feel it is important that we also support automatically opening the a Readme.md file on package install if it's accessible through a link in the package details pane?

I honestly don't know. My request was a knee-jerk reaction to a present lack of functionality; it didn't account for the spec you linked. My immediate thoughts are that I'm not going to know if this is what I want until I play with it.

My scenario is wanting to make documentation obvious to users of company-internal packages that have various (and usually somewhat low) levels of software development sophistication. Making "here's the documentation, in your face" the default is ideal for me. Having it be a link makes it to where the default is "what readme?" instead.

I am wondering if a check-box in the package installer popup (insert correct terminology) that defaults to checked wouldn't be useful. This would make displaying the readme the default, but allow the user to opt-out. Keeping the link in the package details would ensure they can get it if they opt-out and later decide they want it.

Up to y'all how you approach this. I am not familiar with your average user's experience (or your process for that matter), so my perspective might not be ideal to adopt.

Since it's a bigger issue / initiative than I'd thought, the specifics proposed in OP probably aren't the way to go then.

**Comment by chrisraygill** [comment]
Thanks for your feedback! If you wanted to view the documentation for a package you installed, where all would you expect it to be surfaced? 

We've received feedback in the past that the automatic opening of Readme.txt of install can be a jarring context switch for some. Do you not mind it as long as it's useful information?

**Comment by CKraniak** [comment]
Surfacing in the package description area is fine; I like that approach. My feedback is that it requires another quantum of competence to notice / use if it's the only way / replaces the popup, and there are some environments where this matters.

I don't know how the readme context switch is supposed to matter. I'm already in a sort of interstitial, ephemeral context when I'm in the NuGet package installer; is losing focus on the package manager going to impact me all that much? (I'm not saying this perspective is invalid, only that I don't understand it.)

There's some other issues with NuGet opening the readme over and over if it's installed in several projects in a solution. (Might need to be its own issue.) I could see that being unnecessarily distracting.

I think having the documentation popup be switched on/off by the user is a decent approach. My suggestion is either a separate popup on package install asking you if you want the readme (default to yes, have a "don't ask again" checkbox) OR a new (state-saving) checkbox in the "Preview Changes" window.

Thinking out loud: I wonder if you could right-click a package and have a "Show README" in a context menu.

Note: my current approach to the original request is to have a pre-build action that copies the readme.md to readme.txt, and the .txt is in the .nuspec as usual. It's okay enough, since markdown is pretty legible in plain text, and I like having the markdown available for the places where it can be rendered.

**Comment by chrisraygill** [comment]
> Thinking out loud: I wonder if you could right-click a package and have a "Show README" in a context menu.

Do you mean a context menu when right-clicking on a package in the solution explorer? If you have any other thoughts about what experience you would like in a right-click context menu, please feel free to add to the discussion here: https://github.com/NuGet/Home/issues/9007

Currently we are working on a right-click context menu package update experience and would love feedback on that as well other context menu options on this issue: https://github.com/NuGet/Home/issues/9007

**Comment by kikaragyozov** [comment]
Are there any plans on revisiting this?

**Comment by JonDouglas** [comment]
Not until there is some other work done to show readme in tooling experiences.
